pyppeteer/execution_context.py

```diff
@@ -122,117 +109,48 @@ def _convertArgument(self, arg: Any) -> Dict:  # noqa: C901
-
-    async def jsonValue(self) -> Dict:
-        """Get Jsonized value of this object."""
-        objectId = self._remoteObject.get('objectId')
-        if objectId:
-            response = await self._client.send('Runtime.callFunctionOn', {
-                'functionDeclaration': 'function() { return this; }',
-                'objectId': objectId,
-                'returnByValue': True,
-                'awaitPromise': True,
-            })
-            return helper.valueFromRemoteObject(response['result'])
-        return helper.valueFromRemoteObject(self._remoteObject)
-
-    def asElement(self) -> Optional['ElementHandle']:
-        """Return either null or the object handle itself."""
-        return None
-
-    async def dispose(self) -> None:
-        """Stop referencing the handle."""
-        if self._disposed:
-            return
-        self._disposed = True
-        try:
-            await helper.releaseObject(self._client, self._remoteObject)
-        except Exception as e:
-            debugError(logger, e)
-
-    def toString(self) -> str:
-        """Get string representation."""
-        if self._remoteObject.get('objectId'):
-            _type = (self._remoteObject.get('subtype') or
-                     self._remoteObject.get('type'))
-            return f'JSHandle@{_type}'
-        return 'JSHandle:{}'.format(
-            helper.valueFromRemoteObject(self._remoteObject))
-
-
-def _rewriteError(error: Exception) -> None:
-    if error.args[0].endswith('Cannot find context with specified id'):
-        msg = 'Execution context was destroyed, most likely because of a navigation.'  # noqa: E501
-        raise type(error)(msg)
+            raise ElementHandleError('Prototype JSHandle must not be referencing primitive value')
+        response = await self._client.send(
+            'Runtime.queryObjects', {'prototypeObjectId': prototypeHandle._remoteObject['objectId']}
+        )
+        return createJSHandle(context=self, remoteObject=response.get('objects'))
+
+    async def _adoptBackendNodeId(self, backendNodeId: int):
+        obj = await self._client.send(
+            'DOM.resolveNode', {'backendNodeId': backendNodeId, 'executionContextId': self._contextId}
+        )
+        obj = obj['object']
+        return createJSHandle(context=self, remoteObject=obj)
+
+    async def _adoptElementHandle(self, elementHandle: ElementHandle):
+        if elementHandle.executionContext == self:
+            raise ElementHandleError('Cannot adopt handle that already belongs to this execution context')
+        if not self._world:
+            raise ElementHandleError('Cannot adopt handle without DOMWorld')
+        nodeInfo = await self._client.send('DOM.describeNode', {'objectId': elementHandle._remoteObject['objectId']})
+        return await self._adoptBackendNodeId(nodeInfo['node']['backendNodeId'])
+
+
+def rewriteError(error: Exception) -> Union[None, Dict[str, Dict[str, str]]]:
+    msg = error.args[0]
```

Comment:
Issue #119 
We probably want to check if `error.args` has any elements, otherwise this throws an exception

Reply:
I agree! We're accepting PRs against this branch if you'd like to make the necessary changes